README.md

```diff
@@ -1,2 +1,393 @@
+    string queryLowercase = query.toLowerAscii();
+    return product.name.toLowerAscii().includes(queryLowercase) || product.description.toLowerAscii().includes(queryLowercase);
+}
+```
+
+You can read more about Query expressions in this blog. You can have much more complicated queries using limit,let keywords, ordering, joins and so on. You can use query expressions not only for arrays but for streams, and tables as well.
+
+
+## Concurrency safety - Ad Service
+Ballerina is designed for network based applications. The concept of isolation in Ballerina simplifies development by ensuring the safety of shared resources during concurrent execution. Ballerina Compiler warns if the application is not concurrent safe and helps to make it concurrent safe and performant at the same time. The following code shows how a class is marked as readonly so by default compiler makes enables concurrent calls to its objects.
+```ballerina
+readonly class AdStore {
+
+    final map<Ad[]> & readonly ads;
+    private final int MAX_ADS_TO_SERVE = 2;
+
+    isolated function init() {
+        self.ads =  getAds().cloneReadOnly();
+    }
+
+    public isolated function getRandomAds() returns Ad[]|error {
+    }
+}
+```
+You can read this blog for more information about isolation concepts.
+
+## Coordinating with multiple services and Configurables - Checkout Service
+Microservices often requires to communicate with other services to get a specific task done. Checkout service coordinates with cart service, catalog service, currency service, shipping service payment service and email service to perform the checkout. 
+
+```ballerina
+configurable string cartUrl = "http://localhost:9092";
+
+@grpc:ServiceDescriptor {descriptor: ROOT_DESCRIPTOR_DEMO, descMap: getDescriptorMapDemo()}
+service "CheckoutService" on ep {
+    final CartServiceClient cartClient;
+    …
+    
+    function init() returns error? {
+        self.cartClient = check new (cartUrl);
+        …
+    }
+
+    remote function PlaceOrder(PlaceOrderRequest value) returns PlaceOrderResponse|error {
+        string orderId = uuid:createType1AsString();
+        CartItem[] userCartItems = check self.getUserCart(value.user_id, value.user_currency);
+
+           …  
+    }
+}
+
+  function getUserCart(string userId, string userCurrency) returns CartItem[]|error {
+        GetCartRequest req = {user_id: userId};
+        Cart|grpc:Error cart = self.cartClient->GetCart(req);
+        if (cart is grpc:Error) {
+            log:printError("failed to call getCart of cart service", 'error = cart);
+            return cart;
+        }
+        return cart.items;
+    }
+```
+As you shown in above code, ballerina make it very easy to invoke other microservices, log and handle errors. Configurable feature helps to configure the value of the variable by overriding in the runtime. This will be explained in depth in the testing and deployment sections of this article.
```

Comment:
```suggestion
As shown in the above code, Ballerina makes it very easy to invoke other microservices, log, and handle errors. The configurable feature helps to configure the value of the variable by overriding it in the runtime. This will be explained in depth in the testing and deployment sections of this article.
```